### PR TITLE
Fix tifffile and __array_function__ failures in our CI

### LINKED ISSUE #3985: Test failures with NumPy 1.17.0rc1
## Description

Some of our tests are failing with NumPy 1.17.0rc1

https://travis-ci.org/scikit-image/scikit-image/jobs/552547744#L2635

They appear to be related to `__array_function__`.

### PR DESCRIPTION
**Pull request by jni**
## Description

Fixes #3985 #3991 

- Use new `key` keyword argument in Tifffile to replace `pages`
- Use `np.amin` and `np.amax` instead of `min` and `max` to ensure compatibility between NumPy 1.17 (with default `__array_function__` protocol) and various versions of dask.

## Checklist

- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] ~~Gallery example in `./doc/examples` (new features only)~~
- [ ] ~~Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark~~
- [x] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Seems like amin is the correct function in either case.

**Comment by cgohlke** [comment]
```Python
    # parse_kwargs will extract keyword arguments intended for the TiffFile
    # class and remove them from the kwargs dictionary in-place
    tiff_keys = ['multifile', 'multifile_close', 'fastij', 'is_ome']
    kwargs_tiff = parse_kwargs(kwargs, *tiff_keys)


    # read and return tiff as numpy array
    with TiffFile(fname, **kwargs_tiff) as tif:
        return tif.asarray(**kwargs)
```

Does `return tifffile.imread(fname, **kwargs)` not work?

**Comment by jni** [comment]
@cgohlke 

> Does `return tifffile.imread(fname, **kwargs)` not work?

I'm not sure, as I didn't write that plugin. Could it be that the plugin predates that function? Anyway I'll give it a try, thanks!

File: skimage/util/tests/test_arraypad.py [hunk 1]
```diff
@@ -989,6 +991,8 @@ def test_check_kwarg_not_allowed(self):
         with testing.raises(ValueError):
             pad(arr, 4, mode='mean', reflect_type='odd')
 
+    @testing.skipif(LooseVersion(np.__version__) >= LooseVersion('1.17'),
```
Comment (hmaarrfk) [inline skimage/util/tests/test_arraypad.py]: You could just remove this test entirely

**Comment by jni** [comment]
Why on earth is Travis failing now??? Same crap as before, but now with numpy.amin, which *should* work!

**Comment by jni** [comment]
https://travis-ci.org/scikit-image/scikit-image/jobs/555279883#L2806

**Comment by jni** [comment]
Hmmm I misinterpreted what the problem was with dask. np.amin was itself not implemented in dask. Here's the fix:

https://github.com/dask/dask/pull/5043/files

Merged in dask master. I guess I'll go back to using the min/max methods for now.

**Comment by jni** [comment]
... But I don't understand why it fixed appveyor and my local env... =\

**Comment by jni** [comment]
@hmaarrfk regarding removal of the test... Sure but there's a bigger operation hiding there. We should remove our own "wrapper" of pad and crop. I'll just leave all that mess for another day.

**Comment by hmaarrfk** [comment]
Maybe. But whatever works for you. I'm not one to hold up approval for that kind of thing anyway. Looks good.

**Comment by cgohlke** [comment]
>>  Does return `tifffile.imread(fname, **kwargs)` not work?

> Could it be that the plugin predates that function? 

I originally added `imread`, `imsave`, `imshow`, and `_app_show` functions so the tifffile module could be used directly as a skimage plugin. Now `imread` doesn't support the newer `img_num` parameter.

**Comment by sciunto** [comment]
> @hmaarrfk regarding removal of the test... Sure but there's a bigger operation hiding there. We should remove our own "wrapper" of pad and crop. I'll just leave all that mess for another day.

Do you want to open a ticket?

**Comment by sciunto** [comment]
@jni Do you want to backport?

**Comment by jni** [comment]
No, and certainly not until after the last remaining tests are fixed. I think we'll need to coerce to array in that case.

Backports are "by popular demand". I don't think we have many people pining for 0.14.4. The biggest customer is CellProfiler and they are busy upgrading to Python 3. @0x00b1 have I got that right?

**Comment by sciunto** [comment]
Sounds good. We will have to engage the discussion (or I missed it) to define a due date at which we will stop maintaining the LTS

**Comment by jni** [comment]
That date is the Py2.7 end-of-life, Jan 1 2020.

**Comment by sciunto** [comment]
So, we take the same date?

**Comment by hmaarrfk** [comment]
i don't know, i kinda feel like we should backport while we remember to. If 0.14.4 never gets released, so be it, but we will forget that this was a critical fix at some point.

**Comment by stefanv** [comment]
> Backports are "by popular demand". I don't think we have many people pining for 0.14.4. The biggest customer is CellProfiler and they are busy upgrading to Python 3. @0x00b1 have I got that right?

```
| version | python_version | download_month | download_count |
| ------- | -------------- | -------------- | -------------- |
| 0.15.0  | 3.6            | 2019-06        |        878,403 |
| 0.14.2  | 3.6            | 2019-06        |        636,369 |
| 0.14.2  | 3.6            | 2019-07        |        400,205 |
| 0.15.0  | 3.6            | 2019-07        |        336,881 |
| 0.14.2  | 2.7            | 2019-06        |        285,706 |
| 0.15.0  | 3.5            | 2019-06        |        147,212 |
| 0.14.3  | 2.7            | 2019-06        |        121,125 |
| 0.14.2  | 2.7            | 2019-07        |         83,464 |
| 0.14.3  | 2.7            | 2019-07        |         73,112 |
| 0.15.0  | 3.5            | 2019-07        |         63,670 |
| Total   |                |                |      3,026,147 |
```

**Comment by jni** [comment]
@stefanv those numbers don't immediately paint a clear story to me, but anyway, my view of popular demand is that someone stuck on 2.7 requires a specific fix (and expresses this on issues or the mailing list), not number of users.

**Pull request merged**